src/lib/platform/WlClipboardCollection.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Deskflow -- mouse and keyboard sharing utility
+ * SPDX-FileCopyrightText: (C) 2025 Deskflow Developers
+ * SPDX-License-Identifier: GPL-2.0-only WITH LicenseRef-OpenSSL-Exception
+ */
+
+#include "platform/WlClipboardCollection.h"
+
+#include "base/Log.h"
+#include "deskflow/ClipboardTypes.h"
+
+namespace deskflow {
+
+WlClipboardCollection::WlClipboardCollection()
+{
+  initialize();
+}
+
+WlClipboardCollection::~WlClipboardCollection()
+{
+  cleanup();
+}
+
+bool WlClipboardCollection::isAvailable() const
+{
+  return m_available;
+}
+
+IClipboard *WlClipboardCollection::getClipboard(ClipboardID id) const
+{
+  if (!m_available || id >= m_clipboards.size()) {
+    return nullptr;
+  }
+
+  return m_clipboards[id].get();
+}
+
+bool WlClipboardCollection::hasChanged() const
+{
+  if (!m_available) {
+    return false;
+  }
+
+  for (const auto &clipboard : m_clipboards) {
+    if (clipboard && clipboard->hasChanged()) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+void WlClipboardCollection::startMonitoring()
+{
+  if (!m_available || m_monitoring) {
+    return;
+  }
+
+  for (const auto &clipboard : m_clipboards) {
+    if (clipboard) {
+      clipboard->startMonitoring();
+    }
+  }
+
+  m_monitoring = true;
+}
+
+void WlClipboardCollection::stopMonitoring()
+{
+  if (!m_available || !m_monitoring) {
+    return;
+  }
+
+  for (const auto &clipboard : m_clipboards) {
+    if (clipboard) {
+      clipboard->stopMonitoring();
+    }
+  }
+
+  m_monitoring = false;
+}
+
+void WlClipboardCollection::resetChanged() const
+{
+  if (!m_available) {
+    return;
+  }
+
+  for (const auto &clipboard : m_clipboards) {
+    if (clipboard) {
+      clipboard->resetChanged();
+    }
+  }
+}
+
+void WlClipboardCollection::initialize()
+{
+  // Check if Wayland clipboard is available
+  if (!WlClipboard::isAvailable()) {
+    LOG_WARN("wl-clipboard tools not found, clipboard functionality disabled");
+    return;
+  }
```

Comment:
The message is confusing as it's shown when the setting is disabled. This needs to be 2 separate messages; a debug message for the setting and a warning for the missing command.

```suggestion
  if (!WlClipboard::isEnabled()) {
    LOG_DEBUG("wl-clipboard setting disabled");
    return;
  }
  
  if (!WlClipboard::isAvailable()) {
    LOG_WARN("wl-clipboard tools not found, clipboard functionality disabled");
    return;
  }
```